cpp/CMakeLists.txt

```diff
@@ -49,6 +49,8 @@ endif()
 
 set(ARROW_VERSION "11.0.0-SNAPSHOT")
 
+set(CMAKE_EXPORT_COMPILE_COMMANDS 1) # Write compile_commands.json
```

Comment:
I don't think we want to enable this by default. You should pass this yourself to CMake using `-DCMAKE_EXPORT_COMPILE_COMMANDS=1`.

Reply:
OK